src/header/launchbar/LaunchbarIconWithNotification.tsx

```diff
@@ -0,0 +1,61 @@
+/**
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+import classNames from 'classnames';
+
+import LaunchbarButton, { LaunchbarButtonProps } from './LaunchbarButton';
+
+import { JobStatus } from 'src/content/app/tools/blast/state/blast-results/blastResultsSlice';
+
+import styles from './Launchbar.scss';
+
+type LaunchbarButtonWithNotificationProps = {
+  buttonConfig: LaunchbarButtonProps;
+  notificationConfig: {
+    jobStatus: JobStatus;
+    shouldShowNotification: boolean;
+  };
+};
```

Comment:
Something not quite right about these props. A notification launchbar button shouldn't know anything about jobs. What it needs to know is whether it wants to show a notification, and what kind. I suggest we do something like this:

```
type Notification = 'red' | 'green';

type Props = LaunchbarButtonProps & {
  notification: Notification | null;
};
```

(Or you might think of something better than 'red' and 'green'.)

Reply:
✅